aspnet/security/accconfirm.rst

```diff
@@ -1,236 +1,209 @@
+
+Enable SSL in Visual Studio
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. From the **Project** menu, select **Web app Properties**. 
+2. Select **Debug** in the left pane (see image below).
+3. Check **Enable SSL**, and then save changes (this step is necessary to populate the URL box).
+4. Copy the **URL** and paste it into the **Launch URL** box.
+
+.. image:: accconfirm/_static/ssl.png
+
+Require HTTPS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Add the ``[RequireHttps]`` attribute to each controller. The ``[RequireHttps]`` attribute will redirect all HTTP GET requests to HTTPS GET and will reject all HTTP POSTs. A security best practice is to use HTTPS for all requests.
+
+.. literalinclude:: accconfirm/sample/WebApplication1/src/WebApplication1/Controllers/HomeController.cs
+   :language: c#
+   :lines: 9-10
+   :emphasize-lines: 9
+   :dedent: 4
+
+Require email confirmation
+----------------------------
+
+It's a best practice to confirm the email of a new user registration to verify they are not impersonating someone else (that is, they haven't registered with someone else's email). Suppose you had a discussion forum, you would want to prevent "bob@example.com" from registering as "joe@contoso.com". Without email confirmation, "joe@contoso.com" could get unwanted email from your app. Suppose Bob accidentally registered as  "bib@example.com" and hadn't noticed it, he wouldn't be able to use password recover because the app doesn't have his correct email. Email confirmation provides only limited protection from bots and doesn't provide protection from determined spammers, they have many working email aliases they can use to register.
+
+You generally want to prevent new users from posting any data to your web site before they have been confirmed by email, a SMS text message or another mechanism. In the sections below, we will enable email confirmation and modify the code to prevent newly registered  users from logging in until their email has been confirmed.
 
 Setup up Email provider
------------------------
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We'll use the :ref:`Options pattern <options-config-objects>` to access the user account and key settings. For more information, see :doc:`../fundamentals/configuration`.
 
-Although this tutorial only shows how to add email notification through `SendGrid <https://sendgrid.com/>`_, you can send email using SMTP and other mechanisms.
- - Install SendGrid NuGet package
+ - Create a class to fetch the secure email key. For this sample, the ``AuthMessageSenderOptions`` class is created in the *Services/AuthMessageSenderOptions.cs* file.
+
+  .. literalinclude:: accconfirm/sample/WebApplication1/src/WebApplication1/Services/AuthMessageSenderOptions.cs
+   :language: c#
+   :lines: 3-7
+   :dedent: 4
+
+Set the ``SendGridUser`` and ``SendGridKey`` with the `secret manager tool <http://docs.asp.net/en/latest/security/app-secrets.html>`_.
```

Comment:
It would help to show the JSON that you need to add inside secrets.json. I was able to figure it out from reading the Options doc, but an example would make it explicit.